### DISCUSSION TITLE
Linking gitlab issues in changelog

### DISCUSSION THREAD

**Discussion by grzesuav** [goreleaser/goreleaser#4686] [Q&A] [answered]
Hi, not sure if I missed something, however - when I have commit linking to an issue, currently the changelog looks 
![image](https://github.com/goreleaser/goreleaser/assets/9568421/dcd48d27-279b-4970-89f1-4009ceb5ac75)
what I missing here, one of the commit refers an issue in repository. would it be listed also in changelog ? 

Like `closing #4`

What do you think ?

**Comment by caarlos0** [comment, marked as answer]
are you using the gitlab changeloger? if yes, its the response from the api, we don't do anything extra with it.

if you're using the git changeloger, it runs `git log --oneline` basically.

in either case, unfortunately, not much can be done.

---


One thing I looked into in the past was using go-git to parse the `git log`, but it wasn't possible to use it because it was missing a bunch of features (don't remember which exactly).

**Reply by grzesuav** [reply]
Yes, gitlab I guess. Got it.

**Comment by grzesuav** [comment]
@caarlos0 regarding gilab api - for each commit it is possible to get a list of MR's associated with the commit - https://docs.gitlab.com/ee/api/commits.html#list-merge-requests-associated-with-a-commit 
and for each MR - list of issues this MR closes - https://docs.gitlab.com/ee/api/merge_requests.html#list-issues-that-close-on-merge

I believe it would be a way to compose it, what do you think ?

**Reply by caarlos0** [reply]
seems like something that would rate limit pretty quickly if the release has a lot of commits...

not sure if I would want to do this tbh

**Reply by grzesuav** [reply]
understood. need to check how sematic release is handling it
